controllers/profiles/common/object_creators.go

```diff
@@ -157,7 +159,13 @@ func defaultContainer(workflow *operatorapi.SonataFlow) (*corev1.Container, erro
 	if err := mergo.Merge(defaultFlowContainer, workflow.Spec.PodTemplate.Container.ToContainer(), mergo.WithOverride); err != nil {
 		return nil, err
 	}
-	defaultFlowContainer = ConfigurePersistence(defaultFlowContainer, workflow.Spec.Persistence, defaultSchemaName, workflow.Namespace)
+	if workflow.Spec.Persistence != nil {
```

Comment:
Please correct me if I'm wrong, but believe this code is not considering the case where the workflow don't have a configured persistence, but the platform has. In that case, we should try to get it from the platform.

I am correct?

Reply:
There are 3 scenarios managed here:
* Persistence field in the workflow spec is nil: If the persistence field is nil, then the workflow will have no persistence at all. 
* Persistence is empty (`{}`): Persistence will be pull from the platform if it exists.
* Persistence is populated: Use the one in workflow.

Let me know if that makes sense in this case.

Reply:
It makes sense. But what about adding a label to address this problem `sonataflow.org/persistence=platform|workflow|none`, defaults to none.

The problem with using this approach, @jordigilh @tchughesiv is that the API is not communicating what will be done. It's an error prune having `.spec.persistence: {}` meaning that we will pull from the platform.

On the other hand, having `.spec.persistence: <data>` clears communicate that we are overriding. But `.spec.persistence: {}` or not having this attribute defined is dubious IMO. One can interpret that it will pull from platform if the attribute is not there.

Reply:
Actually there is the `enabled` field in the service that is set to true if the service is defined (not nil) by the operator. It's something we're doing already in that sense: define an empty service for DI and the operator will populate it with the default configuration, in this case `enabled=true`.

I'm fine using labels as well. Consistency is important at the end, so if labels/annotations is the path forward for these cases, then this should be the approach accross whenever facing these situations.

Reply:
I'd look at this as:
* No persistence configured: `.spec.persistence: {}`
  * Sets explicitly the empty value for  persistence.
* Use workflow's persistence configuration: `.spec.persistence: <data>`
* Use the system/platform default: `.spec.persistence: nil`
  * By omitting this, the decision is propagated to platform/cluster
  
+1 for consistency (always).

Reply:
The problem is that by default it will always be nil, so it will try to retrieve from the platform. For backwards compatibility `nil` is the best choice as it will not change the behavior in existing workflows and doesn't require the CR to declare the field if no persistence is required.

What @ricardozanini is proposing, in my opinion is not ideal because label/annotations are unstructured data (map of strings), whereas with a field we can enforce validation by defining accepted values, which we own and control. Example is the `imagePullPolicy` in the pod schema. I see  labels/annotations useful when the values can vary depending on the use case, such as the multus network configuration.

Reply:
@masayag @jordigilh using the attributes nil/empty/data to define if we configure a certain feature is error-prone, leading to misconfiguration and user misinterpretation. @masayag comment's to prove my point, he interpreted differently from what you're proposing initially.

The `.spec.dataIndex.enabled` is different since we are forcing an `enabled` attribute to clearly define that we opt-in/out to the feature.

In this case here, labels are well suited since we are using an enum to validate and it clearly expresses the operator's behavior. Also, @wmedvede @tchughesiv any thoughts?

Reply:
imo @masayag's interpretation is how i've generally approached these things and seen them implemented in k8s.

Reply:
@tchughesiv if k8s is using this approach I understand that's a well-known pattern. In this case, +1 to @masayag approach. Can you point me to any other places using this approach?

Reply:
> > if persistence is `nil` in the workflow, settings are used from platform CR.
> 
> Doesn't that break backwards compatibility? I mean, currently a workflow that doesn't have the persistence field populated is not expecting to have persistence if the platform contains one.

what do you mean by `currently `? isn't this PR introducing this capability and we get to define how it functions? how would this break backwards compatibility?

Reply:
Be

> > > if persistence is `nil` in the workflow, settings are used from platform CR.
> > 
> > 
> > Doesn't that break backwards compatibility? I mean, currently a workflow that doesn't have the persistence field populated is not expecting to have persistence if the platform contains one.
> 
> what do you mean by `currently `? isn't this PR introducing this capability and we get to define how it functions? how would this break backwards compatibility?

Because if we define that when persistence is nil, which is the default value, then it should derive the persistence from the platform if it´s defined, then workflows that are currently running without persistence (the ones before merging this PR), will suddently now inherit persistence from the platform even though they don´t require it.

That´s what I mean by backwards compatibility:
* Currently `persistence=nil` means no persistence, it matches the behavior that the workflows expect before this PR is merged
* After this PR is merged, then `persistence=nil` means it might have persistence depending on the platform, whether the workflow requested it or not.

I think that if we have a way to discern if the workflow requires persistence or not, regardless of the value of the persistence struct, we will save ourselves all this trouble and just apply what needs to be applied.

Reply:
> * After this PR is merged, then `persistence=nil` means it might have persistence depending on the platform, whether the workflow requested it or not.

but it won't, because existing platform CRs won't have the `spec.persistence` feild set. it's a new field. hence, existing workflows won't inherit persistence unless the user decides to enable it either within the workflow or the existing platform. we should be good to go, no backwards compat issues.